dev/src/document.ts

```diff
@@ -498,12 +498,17 @@ export class DocumentSnapshot<T = DocumentData> {
    * @private
    */
   toDocumentProto(): api.IDocument {
-    return {
-      createTime: this.createTime,
-      updateTime: this.updateTime,
-      name: this._ref.formattedName,
-      fields: this._fieldsProto!,
-    };
+    const result: api.IDocument = {name: this._ref.formattedName};
```

Comment:
I think you should do this conversion in the caller. The backend has a specified way to return "NoDocuments": https://source.corp.google.com/piper///depot/google3/google/firestore/v1beta1/firestore.proto;l=496

In the calling code, you could do something like 

```
if (doc.exists) ? { document: doc.toProto()} : {name: this._ref.formattedName}
```

In the Mobile SDKs, you want to use as much of the existing deserialization logic as possible. For that, you want to match your Proto format. From looking at your Proto, it seems that you could also add an `exists` to `IBundledDocumentMetadata` and omit the document?

Drive by comment on Proto: We may also want to use "name" for document keys since this is what the backend uses.